### DISCUSSION TITLE
Project 1 Part 3

### DISCUSSION THREAD

**Discussion by Aidan-Myatt** [cooperrc/computational-mechanics#30] [Q&A]
Does anyone know what I am doing wrong here? My guess is that I calculated K wrong, but I am not sure how else you would calculate it other than just solving the equation. For reference I got 0.611. The problem with the graph is that 74 degF is reached around 1.25 hours rather than 2 hours like the problem statement. 

![image](https://user-images.githubusercontent.com/106316906/172061197-7a2a24db-bf8b-4f6e-8bbb-805341a169f0.png)

**Comment by caroline-cook** [comment]
I think you have K right because I got the same thing. I know one thing that may help you is changing the range of time to go from a negative number. You'll see a difference in the two plots then and also be able to answer question c.

**Reply by UniquelyAbbi** [reply]
I also got the same K. Mine looks different from your because I messed up the first 4 lines of the code, so I have to go again.

**Comment by cooperrc** [comment]
> Does anyone know what I am doing wrong here? My guess is that I calculated K wrong, but I am not sure how else you would calculate it other than just solving the equation. For reference I got 0.611. The problem with the graph is that 74 degF is reached around 1.25 hours rather than 2 hours like the problem statement.
> 
> ![image](https://user-images.githubusercontent.com/106316906/172061197-7a2a24db-bf8b-4f6e-8bbb-805341a169f0.png)

Awesome observation!!

You are correct, it doesn't reach the measured temperature at the 2-hr mark. You are _approximating_ the constant $k$ with a finite difference approach, what is a more accurate way to make this calculation?

**Reply by Aidan-Myatt** [reply]
Thank you for the response! I am not sure. Would it be changing dt?

**Reply by cooperrc** [reply]
> Thank you for the response! I am not sure. Would it be changing dt?

changing dt would help for the calculation of $k$

If the ambient temperature is constant, then the heat transfer follows an exponential decay e.g. $T(t) = (T_0 - T_{amb})e^{-kt} + T_{amb}$

If you want the constant, $k$, it would help to either consider the the exponential form or remeasure the temperature at 2-hour+ 5 min or 10 min.

**Comment by feliciones** [comment]
@Aidan-Myatt what is your time_elapsed and delta_t in this code? Did you define delta_t as 2 hours?

**Reply by Aidan-Myatt** [reply]
For me, I had delta_t as 0.1, but I would play around with it to see what gives you the best result. t_elapsed is set to 2 hours.

**Reply by feliciones** [reply]
@Aidan-Myatt What equation did you use for time after death, I'm assuming you did 11am - 0.84 hours. But where'd you get the -.84

**Reply by Aidan-Myatt** [reply]
I played around with plugging different numbers into T_func to give me a value which was 98.6. In my case it was 0.85. I then converted this to my minutes by multiplying it by 60 and then subtracted that number from 11:00 AM. This gave me 10:09 AM.

**Reply by FishMacaron** [reply]
Do either of you think there's a way to do it by changing the equation used to find T to find t instead? I personally attempted to do it that way, I just got stuck because once you get a time (in amount of hours) for 98.6 it's hard to pin down the exact hour like 10:00 AM...
